app/Platform/Controllers/GameTopAchieversController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Controllers;
+
+use App\Community\Enums\AwardType;
+use App\Data\PaginatedData;
+use App\Data\UserData;
+use App\Http\Controller;
+use App\Models\Game;
+use App\Platform\Data\GameData;
+use App\Platform\Data\GameTopAchieverData;
+use App\Platform\Data\GameTopAchieversPagePropsData;
+use App\Platform\Data\PlayerBadgeData;
+use App\Platform\Services\GameTopAchieversService;
+use Illuminate\Http\RedirectResponse;
+use Inertia\Inertia;
+use Inertia\Response as InertiaResponse;
+
+class GameTopAchieversController extends Controller
+{
+    public function index(
+        Game $game,
+        GameTopAchieversService $topAchieversService,
+    ): InertiaResponse|RedirectResponse {
+        $this->authorize('viewAny', [$game]);
+
+        $perPage = 50;
+        $currentPage = (int) request()->input('page', 1);
+
+        // Get total entries to calculate the last page.
+        $totalEntries = $game->players_hardcore;
+        $lastPage = (int) ceil($totalEntries / $perPage);
+
+        // If the current page exceeds the last page, redirect to the last page.
+        if ($currentPage !== 1 && $currentPage > $lastPage) {
+            return redirect()->route('game.top-achievers.index', [
+                'game' => $game->id,
+                'page' => $lastPage,
+            ]);
+        }
+
+        $topAchieversService->initialize($game);
+        $paginatedUsers = $topAchieversService->highestPointEarnersQuery()->paginate($perPage);
+
+        $items = [];
+        $rank = 0;
+        $rankScore = -1;
+        $firstRank = 0;
+        $nextRank = 0;
+        foreach ($paginatedUsers->getCollection() as $playerGame) {
```

Comment:
nit: My VSCode seems very unhappy with `getCollection()` here (getting a red squiggly). The red squiggly goes away if this is changed to `$paginatedUsers->items()`.

Reply:
Looks like the nuance comes down to the underlying types between the two (the one in `BuildGameListAction` isn't complaining for me).

In GameTopAchieversController, `$paginatedUsers` has a type of:
```
\Illuminate\Contracts\Pagination\LengthAwarePaginator $paginatedUsers
```

In BuildGameListAction, `$entries` has a type of:
```
\Illuminate\Pagination\LengthAwarePaginator<\App\Models\Game> $entries
```

One is Contracts\Pagination, whereas the other is just Pagination. Not sure why.